### PR TITLE
feat: update ui 

### PR DESCRIPTION
**Pull request by beyond009**
# Pull Request

## Related Issue

Closes SW-0000 MF-6063 MF-6064 MF-6067

## Checklist

- [ ] I have tested these changes locally.
- [ ] I have reviewed the code changes.
- [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/components/Header/ConnectWalletButton.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    return (
+        <button
+            className=" flex items-center gap-[4px] rounded-[99px] p-[2px] pr-[12px] text-center text-[14px] font-normal leading-[22px] text-neutrals9"
+            style={{ background: 'var(--line-green)' }}
+            onClick={() => {
+                if (!openAccountModal || !openConnectModal) return;
+                isConnected ? openAccountModal() : openConnectModal();
+            }}
+        >
+            {' '}
```
Comment (guanbinrui) [inline src/components/Header/ConnectWalletButton.tsx]:
```suggestion
```

File: tailwind.config.ts [hunk 2]
```diff
@@ -8,6 +8,22 @@ const config: Config = {
     ],
     theme: {
         extend: {
+            colors: {
+                neutrals1: '#FCFCFD',
+                neutrals2: '#F4F5F6',
+                neutrals3: '#E6E8EC',
+                neutrals4: '#B1B5C3',
+                neutrals5: '#B1B5C3',
```
Comment (guanbinrui) [inline tailwind.config.ts:16]: Same as `neutrals4`

File: src/providers/StakeAPI.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import urlcat from 'urlcat';
+
+import { BASE_URL } from '@/constants/index.js';
+import { fetchJSON } from '@/helpers/fetchJSON.js';
+
+interface Response {
+    code: number;
+    data: any;
```
Comment (guanbinrui) [inline src/providers/StakeAPI.ts:8]: Avoid `any`.

File: src/constants/index.ts [hunk 4]
```diff
@@ -23,3 +27,37 @@ export const CACHE_AGE_30_DAYS = 'public, s-maxage=2592000';
...
+export const REWARD = '0xf0c196d1b1489738cda956e994e82ef6897e85bc' as `0x${string}`;
+
+export const BASE_URL = 'https://masknetwork-dev.firefly.land/v1/mask_stake';
+
+export const IS_TESTNET = process.env.NEXT_PUBLIC_TEST;
+
+export const STAKE_MANAGER_CONTRACT = {
+    address: IS_TESTNET ? STAKE_MANAGER_TEST : STAKE_MANAGER,
+    abi: StakeManagerABI,
+    chainId: IS_TESTNET ? 11155111 : 1,
```
Comment (guanbinrui) [inline src/constants/index.ts]: Use constant from `viem` and `wagmi`.

File: src/components/StakeTokenInfo/index.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import MaskIcon from '@/assets/logos/mask.svg';
+
+export function StakeTokenInfo() {
+    return (
```
Comment (guanbinrui) [inline src/components/StakeTokenInfo/index.tsx]: Don't use mock data.

File: src/components/StakeRanking/index.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            <div className="z-10 text-[24px] font-bold text-neutrals2">Staking Ranking</div>
+            <div className="z-10 mt-[64px] flex w-full items-center justify-center">
+                <TopStaker
+                    avatar="https://pbs.twimg.com/profile_images/1745828800531992576/pVBm-qbm_400x400.jpg"
+                    name="ddd009"
+                    amount={100}
+                />
+            </div>
+            <div className="mt-[28px] grid grid-cols-[repeat(auto-fill,minmax(90px,1fr))] gap-[28px]">
+                {new Array(30).fill(0).map((_, index) => (
```
Comment (guanbinrui) [inline src/components/StakeRanking/index.tsx]: Mocked?

File: src/components/StakeModal/StakeTokenInput/UnStake.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
+
+import { InputPanel } from '@/components/StakeModal/StakeTokenInput/InputPanel.js';
+
+export function UnStake() {
+    const [amount, setAmount] = useState('');
+    return (
```
Comment (guanbinrui) [inline src/components/StakeModal/StakeTokenInput/UnStake.tsx]: Mock data

File: src/components/StakeModal/StakeTokenInput/Stake.tsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+                disabled={!amount}
+                className={`flex w-full items-center justify-center rounded-[90px] py-[16px] text-[16px] font-bold text-neutrals8 ${amount ? '' : 'opacity-50'}`}
+                onClick={async () => {
+                    if (allowance < parseEther(amount)) {
+                        const txHash = await writeContract({
+                            ...MASK_TOKEN_CONTRACT,
+                            functionName: 'approve',
+                            args: [STAKE_MANAGER_CONTRACT.address, parseEther(amount)],
+                        });
+                        console.log('txHash', txHash);
```
Comment (guanbinrui) [inline src/components/StakeModal/StakeTokenInput/Stake.tsx]:
```suggestion
```

File: src/components/StakeModal/StakeTokenInput/Stake.tsx [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+                <div>Unlock MASK Time</div>
+                <div className="font-bold text-secondary3">
+                    {poolInfo ? dayjs(poolInfo.endTime).format('HH:MM MM/DD/YYYY') : null}
+                </div>
+            </div>
+            <div className="flex items-center justify-between text-[16px] text-neutrals4">
+                <div>APY</div>
+                <div className="font-bold"> {Math.floor(apyInfo?.data?.apy) || 0}%</div>
+            </div>
+            {/*
```
Comment (guanbinrui) [inline src/components/StakeModal/StakeTokenInput/Stake.tsx]: Remove commented code.

File: src/components/StakeModal/StakeTokenInput/Stake.tsx [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+    const allowance = allowanceFetch || allowanceSet;
+    console.log('allowance', allowance, parseEther(amount));
+    return (
+        <>
+            <InputPanel amount={amount} setAmount={setAmount} balance={balance} />
+            {/* <SelectLockTime setLockTime={setLockTime} lockTime={lockTime} /> */}
+            <div className="flex items-center justify-between text-[16px] text-neutrals4">
+                <div>Unlock MASK Time</div>
+                <div className="font-bold text-secondary3">
+                    {poolInfo ? dayjs(poolInfo.endTime).format('HH:MM MM/DD/YYYY') : null}
```
Comment (guanbinrui) [inline src/components/StakeModal/StakeTokenInput/Stake.tsx]: Should hide the entrie `<div />` if no `pollInfo`.

File: src/components/StakeModal/StakeTokenInput/Stake.tsx [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+            const endTimeStamp = Number(res[1] - res[0]) * 12 + Number(startTimeStamp.timestamp);
+            return {
+                startTime: new Date(Number(startTimeStamp.timestamp) * 1000),
+                endTime: new Date(endTimeStamp * 1000),
+            };
+        },
+    });
+
+    const allowance = allowanceFetch || allowanceSet;
+    console.log('allowance', allowance, parseEther(amount));
```
Comment (guanbinrui) [inline src/components/StakeModal/StakeTokenInput/Stake.tsx]:
```suggestion
```

File: src/components/StakeModal/StakeTokenInput/Stake.tsx [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+    });
+
+    const { data: apyInfo } = useQuery({
+        queryKey: ['poolinfo', 'apy'],
+        queryFn: async () => {
+            const res = stakeAPI.getPoolInfo(POOL_ID);
+            return res;
+        },
+    });
+    const { data: poolInfo } = useQuery({
```
Comment (guanbinrui) [inline src/components/StakeModal/StakeTokenInput/Stake.tsx:54]: Merge queries along a single file. And use `useQuery` once here.

File: src/components/StakeModal/StakeTokenInput/SelectLockTime.tsx [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+                </button>
+                <button
+                    className={`text-[16px] ${lockTime === 2 ? 'bg-neutrals1 text-neutrals8' : 'bg-none text-neutrals2'} flex-1 rounded-[90px] border-[2px] border-neutrals6 px-[16px] py-[12px]`}
+                    onClick={() => {
+                        if (lockTime !== 0) setLockTime(2);
+                    }}
+                >
+                    3 Month
+                </button>
+            </div>
```
Comment (guanbinrui) [inline src/components/StakeModal/StakeTokenInput/SelectLockTime.tsx]: Please refactor:

```tsx
[1,2,3].map(<Button />)
```

File: src/components/StakeCard/index.tsx [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+interface StakeCardProps {
+    setIsOpen: (isOpen: boolean) => void;
+}
+
+export function StakeCard({ setIsOpen }: StakeCardProps) {
+    const { address } = useAccount();
+    const { data, isLoading, isError } = useQuery({
+        queryKey: ['stake'],
+        queryFn: async () => {
+            if (!address) return {};
```
Comment (guanbinrui) [inline src/components/StakeCard/index.tsx]:
```suggestion
            if (!address) return {};
```
```suggestion
            if (!address) return
```

File: src/components/ProjectDetailCard/index.tsx [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+                This is the first phase of the MASK staking event. You will be able to retrieve your staked Mask tokens
+                and reward tokens only after the event concludes.
+            </div>
+            <MainButton
+                className="mt-[24px] w-[255px] !text-neutrals1"
+                onClick={() => {
+                    setIsOpen(true);
+                }}
+                style={{
+                    background: '#05050A',
```
Comment (guanbinrui) [inline src/components/ProjectDetailCard/index.tsx:146]: Use the tailwind way.

File: src/components/ProjectDetailCard/index.tsx [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+                </div>
+                <div
+                    className="flex h-[152px] w-full flex-col items-center gap-[24px] rounded-[16px] border-[1px] border-[white/0.07] bg-[white/0.1] p-[24px] lg:max-w-[350px]"
+                    style={{
+                        boxShadow:
+                            '0px 5px 10px 0px rgba(0, 0, 0, 0.05), 0px 15px 30px 0px rgba(0, 0, 0, 0.05), 0px 20px 40px 0px rgba(0, 0, 0, 0.05)',
+                    }}
+                >
+                    <div className="flex items-center gap-[4px] text-[32px] font-semibold text-neutrals8">
+                        {' '}
```
Comment (guanbinrui) [inline src/components/ProjectDetailCard/index.tsx]:
```suggestion
```

File: src/components/Header/ProfileInfo.tsx [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+            <div className="flex w-full items-center justify-between">
+                <div className="text-[32px] text-neutrals2">Profile</div>
+                <div className="flex items-center justify-center border-[2px] border-neutrals6 bg-neutrals9">
+                    <Close width={24} height={24} />
+                </div>
+            </div>
+            <div className="flex w-full items-center justify-between">
+                <div className="flex items-center gap-[12px]">
+                    <Account />
+                    Makenna Torff
```
Comment (guanbinrui) [inline src/components/Header/ProfileInfo.tsx]: Mock?

File: src/components/Header/NavTabsMobile.tsx [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+    { name: 'FAQs', path: '/faqs' },
+    { name: 'About', path: '/about' },
+];
+
+export function NavTabsMobile() {
+    const pathname = usePathname();
+    return (
+        <nav className="absolute mt-[128px] flex flex-col gap-[36px] self-start lg:hidden" aria-label="Global">
+            {navigation.map((item) => (
+                <div className="flex h-[64px] items-center gap-[30px]">
```
Comment (guanbinrui) [inline src/components/Header/NavTabsMobile.tsx]: Lost `key` here.

File: src/components/Header/NavTabs.tsx [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+];
+
+export interface NavigationHeaderProps {}
+
+export function NavTabs() {
+    const pathname = usePathname();
+    return (
+        <nav className="full ml-[32px] hidden items-center gap-[36px] lg:flex" aria-label="Global">
+            {navigation.map((item) => (
+                <div className="flex flex-col gap-[1px]">
```
Comment (guanbinrui) [inline src/components/Header/NavTabs.tsx]: Lost `key` here.

File: src/components/Footer/BottomBar.tsx [hunk 20]
```diff
@@ -0,0 +1,14 @@
+import { Link } from '@/esm/Link.js';
+
+export function BottomBar() {
+    return (
+        <div className="flex items-center justify-between px-[24px] py-[32px] text-[12px] text-neutrals4">
+            <div>Since 2019 to Now ｜ Mask.io</div>
+            <div className="flex items-center gap-[16px]">
+                <Link href={'#'}>Terms of Use</Link>
+                <Link href={'#'}>Privacy Policy</Link>
+                <Link href={'#'}>Cookie Policy</Link>
```
Comment (guanbinrui) [inline src/components/Footer/BottomBar.tsx:10]: Please consult PM for those links.

File: src/components/Footer/Info.tsx [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+
+const iconsList = [
+    { icon: <Twitter width={40} height={40} />, link: 'https://twitter.com/realmaskbook' },
+    { icon: <Github width={40} height={40} />, link: ' ' },
+    { icon: <Discord width={40} height={40} />, link: ' ' },
+    { icon: <Telegram width={40} height={40} />, link: ' ' },
+    { icon: <Medium width={40} height={40} />, link: ' ' },
+    { icon: <Reddit width={40} height={40} />, link: ' ' },
+    { icon: <Youtube width={40} height={40} />, link: ' ' },
+    { icon: <Facebook width={40} height={40} />, link: ' ' },
```
Comment (guanbinrui) [inline src/components/Footer/Info.tsx:19]: Copy links from `news.mask.io`

File: src/components/EstimatedRewards/index.tsx [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+                    <div className="w-full text-[24px] text-neutrals1">
+                        {Number(rewardPool?.amount).toFixed(2) || 0}
+                    </div>
+                    <div className="w-full text-[16px] text-neutrals1">{rewardToken === 'rss3' ? 'RSS3' : 'TON'}</div>
+                </div>
+            </div>
+            <MainButton
+                onClick={() => {
+                    if (!rewardPool) return;
+                    writeContract({
```
Comment (guanbinrui) [inline src/components/EstimatedRewards/index.tsx:52]: Please display loading state and handle errors.
```suggestion
                    await writeContract({
```

File: src/components/ClaimReward/index.tsx [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+            <div className="flex justify-between">
+                <div className="flex items-center gap-[4px]">
+                    <SLNLogo className="h-[48px] w-[48px]" />
+                    <div className="text-[32px] leading-[48px]">SLN</div>
+                </div>
+                <div className="text-[40px] leading-[48px]">1,300.0002</div>
+            </div>
+            <div className="mt-[12px] flex justify-between text-[14px] text-neutrals9">
+                <div>Reward</div>
+                <div>$ 23.00</div>
```
Comment (guanbinrui) [inline src/components/ClaimReward/index.tsx]: ?

File: src/app/layout.tsx [hunk 24]
```diff
@@ -17,7 +21,13 @@ export default function RootLayout({
 }>) {
     return (
         <html lang="en">
-            <body className={inter.className}>{children}</body>
+            <body className={inter.className + ' relative'}>
```
Comment (nuanyang233) [inline src/app/layout.tsx]: You can use the classNames function or use template strings.

File: src/app/page.tsx [hunk 25]
```diff
@@ -1,111 +1,32 @@
-import { Image } from '@/esm/Image.js';
+import { redirect } from 'next/navigation.js';
+
+import { ProjectCard } from '@/components/ProjectCard/index.js';
+import { StakeRanking } from '@/components/StakeRanking/index.js';
 
 export default function Home() {
+    redirect('/stake');
```
Comment (nuanyang233) [inline src/app/page.tsx:7]: ？

**Review by nuanyang233 — CHANGES_REQUESTED** [review]

File: src/components/StakeCard/index.tsx [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+import { Image } from '@/esm/Image.js';
+
+interface StakeCardProps {
+    setIsOpen: (isOpen: boolean) => void;
+}
+
+export function StakeCard({ setIsOpen }: StakeCardProps) {
+    const { address } = useAccount();
+    const { data, isLoading, isError } = useQuery({
+        queryKey: ['stake'],
```
Comment (swkatmask) [inline src/components/StakeCard/index.tsx]:
```suggestion
        queryKey: ['stake', address],
```
should response to adderss

File: src/components/StakeModal/StakeTokenInput/Stake.tsx [hunk 27]
```diff
@@ -0,0 +1,137 @@
...
+import { IS_TESTNET, MASK_TOKEN_CONTRACT, POOL_ID, STAKE_MANAGER_CONTRACT } from '@/constants/index.js';
+import { stakeAPI } from '@/providers/StakeAPI.js';
+
+export function Stake() {
+    const [amount, setAmount] = useState('');
+    const { address } = useAccount();
+    const [allowanceSet, setAllowanceSet] = useState(BigInt(0));
+
+    const { data: allowanceFetch } = useSuspenseQuery({
+        queryKey: ['allowance'],
```
Comment (swkatmask) [inline src/components/StakeModal/StakeTokenInput/Stake.tsx]:
```suggestion
        queryKey: ['allowance', address],
```

ditto

File: src/components/StakeModal/StakeTokenInput/Stake.tsx [hunk 28]
```diff
@@ -0,0 +1,137 @@
...
+                ...MASK_TOKEN_CONTRACT,
+                functionName: 'allowance',
+                args: [address, STAKE_MANAGER_CONTRACT.address],
+            });
+            return res;
+        },
+    });
+
+    const { data: balance } = useQuery({
+        queryKey: ['balance'],
```
Comment (swkatmask) [inline src/components/StakeModal/StakeTokenInput/Stake.tsx]:
```suggestion
        queryKey: ['balance', address],
```
ditto

**Review by swkatmask — CHANGES_REQUESTED** [review]

File: src/components/ProjectDetailCard/index.tsx (hunk 15 above)
Comment (beyond009) [reply src/components/ProjectDetailCard/index.tsx:146]: tailwind will be overwritten by linear gradient background

**Pull request closed without merging**